com.unity.shadergraph/Editor/Importers/ShaderGraphImporter.cs

```diff
@@ -115,7 +115,32 @@ public override void OnImportAsset(AssetImportContext ctx)
 #endif
             {
                 var text = GetShaderText(path, out configuredTextures, sourceAssetDependencyPaths, graph);
-                shader = ShaderUtil.CreateShaderAsset(text, false);
+
+#if UNITY_2021_1_OR_NEWER
+                // 2021.1 or later is guaranteed to have the new version of this function
+                shader = ShaderUtil.CreateShaderAsset(ctx, text, false);
+#else
+                // earlier builds of Unity may or may not have it
+                // here we try to invoke the new version via reflection
+                var createShaderAssetMethod = typeof(ShaderUtil).GetMethod(
+                    "CreateShaderAsset",
+                    System.Reflection.BindingFlags.Static | System.Reflection.BindingFlags.Public | System.Reflection.BindingFlags.ExactBinding,
+                    null,
+                    new Type[] { typeof(AssetImportContext), typeof(string), typeof(bool) },
+                    null);
```

Comment:
Note: the reflection used here is not properly captured in the import context dependencies...  this means it is possible that if you upgrade ShaderGraph first (invoking this import code), then later upgrade the Unity editor from to 2020.2.0b1 or later, the import system will not realize it needs to re-run all of the ShaderGraph imports to check for changes in reflection.  I don't see any way around this in our current import system though; the only workaround is to re-import all ShaderGraphs.

Reply:
I think better just keep it simple and ensure that we have solid behaviour from 2020.1 -> 2020.2 beta and up.
So just bump the importer version based on ifdef

Reply:
This should only be hit going from 2020.2 alpha -> beta though, so it seems minor enough to not handle this case.

Reply:
Although FYI -- I learned about AssetDatabase.RegisterCustomDependency(); which can let you express complex global dependencies like this... :)

Reply:
No need to bump based on ifdef, as this version of the package only works with 2020.2 anyways, and this PR bumps the version, so anyone coming from 2020.1 will get a reimport.